ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/operators/ProjectOperator.java

```diff
@@ -207,8 +135,34 @@ public AbstractPhysicalOperator getChild(final int index) {
     throw new UnsupportedOperationException();
   }
 
-  public LogicalSchema getOutputSchema() {
-    return outputSchema;
+  private Function<TableRow, GenericRow> getPreSelectTransform() {
```

Comment:
Maybe this should be called getIntermediateRow to match the terminology of the logical node?